tests/macros/test-svginfo.js

```diff
@@ -0,0 +1,496 @@
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'image'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>image</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "marker": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'marker'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>marker</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "mask": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'mask'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>mask</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "pattern": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'pattern'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>pattern</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "script": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'script'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>script</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "style": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'style'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>style</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "switch": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'switch'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>switch</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "text": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'text'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>text</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        },
+        "view": {
+            url: ['/zh-CN', SVG_BASE_SLUG, 'Element', 'view'].join('/'),
+            data: {
+                summary: "这是SVG <strong><code>view</code></strong>元素的模拟",
+                tags: ["SVG", "Element"]
+            }
+        }
+    }
+};
+
+
+// Test cases definition
+// ----------------------------------------------------------------------------
+// Each test case is define by:
+// A `title` to make the test understandable by a human behing
+// An `input` which is an Array of parameters that will be passed to the macro
+// An `output` which is the string that the macro should return,
+// A `env` that is overriding the default env variable inside the macro
+//
+// NOTE: we could probably make that more generic by having a single test
+//       runner (see below) and a bunch of JSON files (one per macro) to
+//       describe all the possible inputs and their expected outputs.
```

Comment:
On further reflection, in order for this approach to be used for more general testing, the definition of each test case would also have to include its `beforeEachMacro` set-up (creating and populating the stubs for example), which seems difficult to standardize.

Reply:
Indeed, at that point I need to get into more testing to see what can be standardize (and how) and what cannot and how is it possible to neatly handle special edge case. Those notes are here to remember to have further discussion when we will have more tests.